gnat2goto/driver/tree_walk.adb

```diff
@@ -2878,7 +2878,16 @@ package body Tree_Walk is
                      Variant_Disc_Value := Disc_Actual;
                   end if;
 
-                  New_Expr := Do_Expression (Disc_Actual);
+                  if Present (Disc_Actual) then
+                     New_Expr := Do_Expression (Disc_Actual);
+                  else
+                     New_Expr :=
+                       Make_Constant_Expr (Source_Location => Sloc (E),
+                                           I_Type          =>
+                                             Do_Type_Reference (Etype (Iter)),
+                                           Range_Check     => False,
+                                           Value           => "0");
```

Comment:
Looks like a dummy value? Shouldn't this be an unsupported feature report or an exception, something of the sort then?

Reply:
Yes. Or better yet, we should find out how to properly initialise the struct-member so that we don't end up calling `Discriminant_Default_Value` on a node that doesn't have one.